.buildkite/pipeline.yml

```diff
@@ -62,3 +61,13 @@ steps:
           - step: "int-test"
             allow_failure: true
     depends_on: "check"
+
+  - label: "Publish docker image"
+    key: "publish"
+    command: ".buildkite/scripts/build_push_docker_image.sh"
+    if: build.env("BUILDKITE_PULL_REQUEST") != null
```

Comment:
I think it is needed to ensure which branches or tags are being executed in Buildkite. since by default `build_branches` is enabled.

And for instance this branch was run in Buildkite (without PR):
- Branch of this PR: https://github.com/elastic/fleet-server/pull/2566
- Buildkite job for the branch event: https://buildkite.com/elastic/fleet-server/builds?branch=updatecli_28b583ca5753aa9b9dea5ffba70409e7bc6ea043c84ea6c3dee72907d8c6e04a
- Buildkite job for the PR event (bot): https://buildkite.com/elastic/fleet-server/builds?branch=elastic:updatecli_28b583ca5753aa9b9dea5ffba70409e7bc6ea043c84ea6c3dee72907d8c6e04a 

cc @alexsapran I think we should try to limit what it is run in Buildkite to the branches to those that are protected (main, 8.* and 7.*) and the corresponding tags.

Catalog-info: 
https://github.com/elastic/fleet-server/blob/main/catalog-info.yaml#L27-L51

It should be added `branch_configuration` as in elastic-package or package-registry to just allow main, 8.* or 7.* branches or v8.* , v7.* tags 

Example: https://github.com/elastic/package-registry/blob/main/catalog-info.yaml#L25



Currently, all

Reply:
Should we change that in this PR, or we can move on next PR?
Because it's changing the logic of the current Jenkins pipeline...

Reply:
It could be done in another Pull Request. This is unrelated to the step defined here in this PR. Just realized about these Buildkite builds while looking at the options of the filtering.

IIRC the required change should be updating the [`catalog-info.yml` file](https://github.com/elastic/fleet-server/blob/main/catalog-info.yaml#L27-L51) to add `branch_configuration` setting with the corresponding values (to be checked the regex to be used).

Reply:
Created PR for this https://github.com/elastic/fleet-server/pull/2624